src/back/spv/writer.rs

```diff
@@ -189,7 +189,16 @@ impl Writer {
             *e.get()
         } else {
             match lookup_ty {
-                LookupType::Handle(handle) => self.write_type_declaration_arena(arena, handle),
+                LookupType::Handle(handle) => match &arena[handle].inner {
+                    crate::TypeInner::Scalar { kind, width } => self.get_type_id(
+                        arena,
+                        LookupType::Local(LocalType::Scalar {
+                            kind: *kind,
+                            width: *width,
+                        }),
+                    ),
+                    _ => self.write_type_declaration_arena(arena, handle),
```

Comment:
it's a bit confusing that `get_type_id` actually writes down the type declaration. Maybe `resolve_type_id` to stress on the mutable action here?

Reply:
Yes that is true, but I think that we need to consider this for the whole back-end as I've used this terminology everywhere unfortunately.

I suggest to postpone it, and pick it up at a later stage.